solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -829,6 +824,7 @@ private NamedList<Object> processErrorsAndResponse(
         // no processor specified, return raw stream
         NamedList<Object> rsp = new NamedList<>();
         rsp.add("stream", is);
+        rsp.add("responseStatus", httpStatus);
```

Comment:
Maybe we should name this "httpStatus" and should use an integer so as to keep generic of the type of HttpClient we are using?

Reply:
this is an int, check a few lines above `int httpStatus = response.getStatus();` and yes this was exactly what I was trying to do, separate implementation details from the response :)